src/main/java/de/hysky/skyblocker/skyblock/dwarven/CorpseFinder.java

```diff
@@ -0,0 +1,269 @@
+                }}}
+        LOGGER.warn(PREFIX + "Couldn't match a color! Something probably went very wrong!");
+        return Formatting.YELLOW;
+    }
+
+    @SuppressWarnings("DataFlowIssue")
+    private static float[] getColors(Formatting color) {
+        return ColorUtils.getFloatComponents(color.getColorValue());
+    }
+
+    private static void parseCords(Text text) {
+        Pattern cordsPattern = Pattern.compile("x: (?<x>-?\\d+), y: (?<y>-?\\d+), z: (?<z>-?\\d+)(?:.+)?");
+        String message = text.getString();
+        Matcher matcher = cordsPattern.matcher(message);
+        if (matcher.find()) {
+            int x = Integer.parseInt(matcher.group("x"));
+            int y = Integer.parseInt(matcher.group("y"));
+            int z = Integer.parseInt(matcher.group("z"));
+            LOGGER.info(PREFIX + "Parsed message! X:{}, Y:{}, Z:{}", x, y, z);
+            boolean foundCorpse = false;
+            BlockPos parsedPos = new BlockPos(x-1, y-1, z-1);   // skyhanni cords format difference is -1, -1, -1
+            for (List<Corpse> corpses : corpsesByType.values()) {
+                for (Corpse corpse : corpses) {
+                    if (corpse.waypoint.pos.equals(parsedPos)) {
+                        corpse.seen = true;
+                        foundCorpse = true;
+                        LOGGER.info(PREFIX + "Setting corpse {} as seen!", corpse.entity);
+                        MinecraftClient.getInstance().player.sendMessage(
+                            Constants.PREFIX.get()
+                                .append("Parsed message from chat, adding corpse at ")
+                                .append(corpse.entity.getBlockPos().up(0).toShortString()));
+                    }}}
+            if (!foundCorpse) {
+                LOGGER.warn(PREFIX + "Did NOT found any match for corpses! corpsesByType.values(): {}", corpsesByType.values());
+                LOGGER.info(PREFIX + "Proceeding to iterate over all corpses!");
+                for (List<Corpse> corpses : corpsesByType.values()) {
+                    for (Corpse corpse : corpses) {
+                        LOGGER.info(PREFIX + "Corpse: {}, BlockPos: {}", corpse.entity, corpse.entity.getBlockPos());
+                    }
+                }
+            }
+        }
+    }
+
+	static class Corpse {
+		private final ArmorStandEntity entity;
+		private final Waypoint waypoint;
+		private boolean opened;
+		private boolean seen;
+		private long messageLastSent = 0;
+		private Formatting color = Formatting.YELLOW;
+		private float[] colors;
+		private String name;
+
+		Corpse(ArmorStandEntity entity, Waypoint waypoint, boolean opened) {
+			this.entity = entity;
+			this.waypoint = waypoint;
+			this.opened = opened;
+			this.seen = false;
+			this.color = getColor(entity);
+			this.colors = getColors(color);
+			this.name = getType(entity);
+		}
+		Corpse(ArmorStandEntity entity, Waypoint waypoint, boolean opened, Formatting color) {
```

Comment:
This  constructor is unused. why dose it still exist?

Reply:
Done in [2294e81](https://github.com/SkyblockerMod/Skyblocker/pull/960/commits/2294e81dc04676c5654c5c0875e38beb97d2cc1d).